### PR TITLE
Migration/l10n br purchase

### PR DESCRIPTION
**Pull request by renatonlima**

### CODE REVIEW

**Comment by rvalyi** [comment]
will need a rebase once l10n_br_stock is merged. But we made tests locally with a local l10n_br_stock copy meanwhile and fixed the code so reviews can already start.

**Comment by rvalyi** [comment]
rebased on 8.0, that is including the l10n_br_stock dependency. However we got again a strange "heisenbug" in the l10n_br_account module that seems totally unrelated to this PR https://travis-ci.org/OCA/l10n-brazil/jobs/92494378#L786
@renatonlima this one is driving us crazy...

**Comment by mileo** [comment]
This is the error that @lfdivino had 9 days ago

https://github.com/OCA/l10n-brazil/pull/134#issuecomment-156265177

**Comment by rvalyi** [comment]
@mileo @lfdivino yes, we never had it at the time he reported it and couldn't find any explanations for it. But then we had it a couple a of times without being able to find any explanation. Seems same code will often pass and fail in rare exceptions. Any help for finding the cause of the bug or a workaround is welcome. I wonder if some database lock may cause a workflow transaction to fail inconsistently or something like that...

**Comment by renatonlima** [comment]
@rvalyi @mileo @lfdivino @mbcosta

I found the error and the problem is in travis all modules are installed and after run all tests for all modules in branch, and now we have a conflict in account.invoice workflow, at moment I change all tests in l10n_br_account to do not confirm any invoice until migration is finish I made a PR to solve this https://github.com/OCA/l10n-brazil/pull/242

**Comment by rvalyi** [comment]
Travis green; @mileo @danimaribeiro reviews welcome...

**Comment by mileo** [comment]
@kmee ! Review please

**Comment by rvalyi** [comment]
:+1: LGTM

**Comment by mileo** [comment]
:+1:

File: l10n_br_purchase/__openerp__.py [hunk 1]
```diff
@@ -1,44 +1,44 @@
...
+        'views/res_company_view.xml',
         'security/ir.model.access.csv',
         'security/l10n_br_purchase_security.xml',
     ],
     'demo': [
-        'test/purchase_order_demo.yml'
-            ],
-    'installable': False,
+        # FIXME
+        # 'test/purchase_order_demo.yml'
```
Comment (mileo) [inline l10n_br_purchase/__openerp__.py:40]: Is this right?

**Comment by renatonlima** [comment]
Yes @mileo, at moment there is not test on purchase, I working in some tests in sale modules, after I will complete this test in purchase.

**Comment by mileo** [comment]
Sale module we cant make tests pass here. =(
How is the way to do it with web_context_tunnel?

**Comment by renatonlima** [comment]
You can define context values in yml like this https://github.com/OCA/l10n-brazil/blob/8.0/l10n_br_purchase/test/purchase_order_demo.yml#L5

**Comment by mileo** [comment]
We are testing the https://github.com/OCA/l10n-brazil/pull/148 with the same approach but does not works well.

**Pull request merged**